frontend/src/containers/DonatePage.js

```diff
@@ -1,224 +1,219 @@
+                  {collective.mission ? <div className='DonatePage-line2'>{ collective.mission }</div> : null}
+                </div>
+            }
+            { view === 'default' &&
+              <Tiers
+                tiers={tiers}
+                collective={collective}
+                host={host}
+                donationForm={donationForm}
+                appendDonationForm={appendDonationForm}
+                onToken={donateToGroup.bind(this)}
+                i18n={i18n}
+              />
+            }
+            { view === 'signup' &&
+              <PostDonationSignUp {...this.props} save={saveNewUser.bind(this)} {...this.props} />
+            }
+            { view === 'thankyou' &&
+              <PostDonationThanks i18n={i18n} collective={collective} message={i18n.getString('thankyou')} tweet={tier.tweet} showRelated={false} closeDonationFlow={this.closeDonationFlow.bind(this)} />
+            }
+            { view === 'default' &&
+              <div className="LightFooter">
+                <a href="/">
+                  <div className="logo">
+                    <svg width='14' height='14' className='inline-block align-middle mr1'>
+                      <use xlinkHref='#svg-isotype' fill='#7FADF2'/>
+                    </svg>
+                    <svg width='138' height='20' className='inline-block align-middle'>
+                      <use xlinkHref='#svg-logotype' fill='#919699'/>
+                    </svg>
+                  </div>
+                </a>
+              </div>
+            }
+          </div>
+        </div>
       </div>
     );
   }
 
+  closeDonationFlow() {
+    const  { collective } = this.props;
+    window.location = `/${collective.slug}`;
+  }
+
   componentWillMount() {
     const {
-      group,
-      fetchUsers,
-      fetchGroup
+      collective,
+      fetchUsers
     } = this.props;
-
-    fetchGroup(group.slug);
-    fetchUsers(group.slug);
+    return fetchUsers(collective.slug);
   }
 
   componentDidMount() {
-    // decode here because we don't handle auth on the server side yet
-    this.props.decodeJWT();
+    const { collective, host } = this.props;
+    console.log("componentDidMount>>> collective", collective, "host", host);
   }
+
 }
 
 export function donateToGroup({amount, interval, currency, description, token}) {
   const {
     notify,
     donate,
-    group,
-    fetchGroup,
-    fetchUsers
+    collective
   } = this.props;
 
   const payment = {
     stripeToken: token && token.id,
     email: token && token.email,
     amount,
-    interval,
     currency,
     description
   };
 
-  return donate(group.id, payment)
-    .then(({json}) => {
-      if (json && !json.hasFullAccount) {
-        this.setState({ showUserForm: true })
-      } else {
-        this.setState({ showThankYouMessage: true })
-      }
-    })
-    .then(() => fetchGroup(group.id))
-    .then(() => fetchUsers(group.id))
+  if (interval !== 'one-time')
```

Comment:
since we aren't storing one-time interval anywhere, will be better in the long run to treat 'interval' as 'monthly' or 'yearly' and if it's missing, we assume it's one-time